tap_chargebee/client.py

```diff
@@ -12,13 +12,122 @@
 LOGGER = singer.get_logger()
 
 
-class Server4xxError(Exception):
+class ChargebeeError(Exception):
     pass
 
+class Server4xxError(ChargebeeError):
+    pass
+
+class Server429Error(ChargebeeError):
+    pass
+
+class Server5xxError(ChargebeeError):
+    pass
+
+class ChargebeeBadRequestError(Server4xxError):
+    pass
+
+class ChargebeeUnauthorizedError(Server4xxError):
+    pass
 
-class Server429Error(Exception):
+class ChargebeeForbiddenError(Server4xxError):
     pass
 
+class ChargebeeNotFoundError(Server4xxError):
+    pass
+
+class ChargebeeMethodNotAllowedError(Server4xxError):
+    pass
+
+class ChargebeeNotProcessedError(Server4xxError):
+    pass
+
+class ChargebeeRateLimitError(Server429Error):
+    pass
+
+class ChargebeeInternalServiceError(Server5xxError):
+    pass
+
+class ChargebeeServiceUnavailableError(Server5xxError):
+    pass
+
+
+STATUS_CODE_EXCEPTION_MAPPING = {
+    400: {
+        "raise_exception": ChargebeeBadRequestError,
+        "message": "The request URI does not match the APIs in the system.",
+    },
+    401: {
+        "raise_exception": ChargebeeUnauthorizedError,
+        "message": "The user is not authorized to use the API.",
+    },
+    403: {
+        "raise_exception": ChargebeeForbiddenError,
+        "message": "The requested operation is not permitted for the user.",
+    },
+    404: {
+        "raise_exception": ChargebeeNotFoundError,
+        "message": "The requested resource was not found.",
+    },
+    405: {
+        "raise_exception": ChargebeeMethodNotAllowedError,
+        "message": "The HTTP action is not allowed for the requested REST API.",
+    },
+    409: {
+        "raise_exception": ChargebeeNotProcessedError,
+        "message": "The request could not be processed because of conflict in the request.",
+    },
+    429: {
+        "raise_exception": ChargebeeRateLimitError,
+        "message": "You are requesting to many requests.",
+    },
+    500: {
+        "raise_exception": ChargebeeInternalServiceError,
+        "message": "The request could not be processed due to internal server error.",
+    },
+    503: {
+        "raise_exception": ChargebeeServiceUnavailableError,
+        "message": "The request could not be processed due to temporary internal server error.",
+    },
+}
+
+
+def get_exception_for_status_code(status_code):
+    """Map the input status_code with the corresponding Exception Class \
+        using 'STATUS_CODE_EXCEPTION_MAPPING' dictionary."""
+
+    exception = STATUS_CODE_EXCEPTION_MAPPING.get(status_code, {}).get(
+                "raise_exception")
+    # If exception is not mapped for any code then use Server4xxError and Server5xxError respectively
+    if not exception:
+        if status_code > 400 and status_code < 500:
+            exception = Server4xxError
+        elif status_code > 500:
+            exception = Server5xxError
+        else:
+            exception = ChargebeeError
```

Comment:
Can you please add a unit test case for these else condition?

Reply:
Added unit tets to verify above else condition too.